### PR TITLE
update(utils): updates to zonbudget and get_specific_discharge

### LINKED ISSUE #1340: `get_specific_discharge` returns discharge at the centers while position='faces'
When I call the function `flopy.utils.postprocessing.get_specific_discharge()` with `position='faces'` I get the discharge of the cell centers instead of the cell faces. Calling the function with `position='centers'` I get exactly the same results.

I tried to use this code to get the specific discharge at the cell faces:
```
cbcobj = flopy.utils.CellBudgetFile(os.path.join(gwf.model_ws, f"{gwf.name}.cbc"))
spdis = cbcobj.get_data(text="DATA-SPDIS")[0]
qx, qy, qz = flopy.utils.postprocessing.get_specific_discharge(spdis, gwf, position="faces")
```

Looking into the `get_specific_discharge` function I can see that the `position` variable is not used when `vertices` is a recarray. In my case `spdis` is a recarray.

It would be nice if the function can return the specific discharge on the cell faces when `vertices` is a recarray. If that is hard or impossible I think an error should be raised when calling this function with a recarray and `position='faces'`.

My modelgrid is a locally refined grid with type 'vertex', for all that matters.

### PR DESCRIPTION
**Pull request by jlarsen-usgs**
* update read_zone_file() to allow for extraneous text after zone arrays are read
* update get_specific_discharge(); add error for "faces" and "vertices" option

#1433 
Closes #1340

### CODE REVIEW

File: examples/data/mf6-freyberg/bot.asc [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
-0.0000 8.8675 10.9370 5.9463 6.7191 6.5623 5.4118 3.4439 2.3840 1.9104 0.5075 0.6844 0.1888 0.4654 0.7109 0.3407 0.1594 1.4753 1.0008 2.4790
-0.0000 0.0000 10.9510 6.1613 6.3923 6.7709 2.9707 3.0016 3.1640 0.9274 0.1631 0.6601 0.2656 0.3331 0.6923 0.4834 0.6389 1.3737 1.6797 1.4661
-0.0000 0.0000 0.0000 7.3548 5.1731 4.9708 5.2845 6.2087 3.1052 1.3327 0.8008 0.3992 0.4791 0.1835 0.5879 0.7434 0.8502 1.9328 1.8556 2.0022
-0.0000 0.0000 0.0000 0.0000 4.2779 4.8438 5.5892 5.0886 3.2245 1.6683 0.6847 0.4031 0.5131 0.5307 0.2503 0.6330 0.6843 1.4369 1.6174 0.0000
-0.0000 0.0000 0.0000 0.0000 5.8926 6.2475 7.1632 3.0387 2.2536 1.6464 0.3108 -0.2027 0.5490 1.0779 0.0649 0.4217 0.3124 1.7811 1.4466 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 6.5855 4.7402 1.7855 2.9122 1.6155 0.7083 0.4167 0.5610 0.4744 0.3715 0.3287 0.4958 1.4285 0.0000 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 5.1503 5.8960 2.3450 2.7254 1.4189 0.1220 0.4666 0.4944 0.1662 0.6749 0.6819 -0.0017 0.0000 0.0000 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 6.9138 4.5808 2.5275 2.6038 1.4444 0.5365 0.2253 0.5896 0.6079 0.5060 0.0000 0.0000 0.0000 0.0000 0.0000
+NCOLS 20
```
Comment (jdhughes-usgs) [inline examples/data/mf6-freyberg/bot.asc:1]: Are these just line ending changes?

File: examples/data/mf6-freyberg/k11.asc [hunk 2]
```diff
@@ -1,46 +1,46 @@
...
-0.0000 0.0001 0.0001 0.0001 0.0001 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0001
-0.0000 0.0000 0.0001 0.0001 0.0001 0.0001 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0000 0.0001 0.0000 0.0001 0.0001 0.0001
-0.0000 0.0000 0.0000 0.0001 0.0001 0.0001 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0000 0.0000 0.0001 0.0001
-0.0000 0.0000 0.0000 0.0000 0.0001 0.0001 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0000 0.0000 0.0000 0.0001 0.0001 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0001 0.0001 0.0001 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0001 0.0000 0.0001 0.0001 0.0001 0.0001 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0001 0.0000 0.0001 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0001 0.0001 0.0000 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0001 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0000 0.0000 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 0.0001 0.0001 0.0001 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000 0.0000
+NCOLS 20
```
Comment (jdhughes-usgs) [inline examples/data/mf6-freyberg/k11.asc:1]: Line ending changes?

File: examples/data/mf6-freyberg/top.asc [hunk 3]
```diff
@@ -1,46 +1,46 @@
...
-0.0000 0.0000 25.9003 25.5183 25.1175 24.6674 24.1454 23.6011 22.7846 21.9279 20.9036 19.5506 17.6990 16.0786 14.6397 15.5185 16.2272 16.7720 17.0695 17.2266
-0.0000 0.0000 0.0000 25.0825 24.5310 24.0663 23.5984 23.1007 22.4714 21.6958 20.5893 19.0353 17.2143 15.8583 14.4106 15.3139 15.9899 16.5181 16.8330 17.0850
-0.0000 0.0000 0.0000 0.0000 23.7381 23.3648 22.9378 22.4520 21.9098 21.1388 19.9832 18.6560 17.0354 15.5765 14.1908 15.1187 15.7983 16.2660 16.5142 0.0000
-0.0000 0.0000 0.0000 0.0000 23.2295 22.5662 22.0596 21.5957 21.1462 20.4836 19.4587 18.3429 16.9059 15.3844 13.9596 14.9805 15.6024 16.0389 16.3466 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 21.2474 20.9537 20.6163 20.3183 19.7002 18.7647 17.8460 16.6769 15.2479 13.7210 14.8247 15.3862 15.8014 0.0000 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 20.0273 19.8268 19.5266 19.3028 18.5765 17.9346 17.1333 16.1436 14.9701 13.5055 14.6651 15.1257 0.0000 0.0000 0.0000
-0.0000 0.0000 0.0000 0.0000 0.0000 18.9000 18.4000 18.1000 17.6000 17.1000 17.1000 16.0000 15.0000 14.5000 14.0000 0.0000 0.0000 0.0000 0.0000 0.0000
+NCOLS 20
```
Comment (jdhughes-usgs) [inline examples/data/mf6-freyberg/top.asc:1]: line ending changes?

**Review by jdhughes-usgs — COMMENTED** [review]
The changes in the 3 `.asc` files look like they just may be line ending changes? If so and the changes are not required for any tests to pass would probably be better to revert them to the original files in the PR.

File: examples/data/mf6-freyberg/top.asc (hunk 3 above)
Comment (jlarsen-usgs) [reply examples/data/mf6-freyberg/top.asc:1]: Yes. I'm having trouble reverting these, but will put in a new PR without these changes. These files are written to examples directory during the autotesting, which I'll examine and update so this doesn't continue to happen.

**Pull request closed without merging**